cmd/juno/cli/root.go

```diff
@@ -30,41 +36,80 @@ var (
 	//go:embed long.txt
 	longMsg string
 
+	processHandler *process.Handler
 	// rootCmd is the root command of the application.
 	rootCmd = &cobra.Command{
 		Use:   "juno [options]",
 		Short: "Starknet client implementation in Go.",
-		Long:  longMsg,
-		Run: func(cmd *cobra.Command, args []string) {
-			handler := process.NewHandler()
+		Run: func(_ *cobra.Command, _ []string) {
+			processHandler = process.NewHandler()
 
 			// Handle signal interrupts and exits.
 			sig := make(chan os.Signal)
 			signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
 			go func() {
 				<-sig
 				log.Default.Info("Trying to close...")
-				handler.Close()
-				log.Default.Info("App closing...Bye!!!")
+				cleanup()
 				os.Exit(0)
 			}()
 
+			// Breaking initial cases
+			if config.Runtime.Ethereum.Node == "" || config.Runtime.Starknet.FeederGateway == "" {
+				log.Default.Panic("Ethereum client needed")
+			}
+
 			// Subscribe the RPC client to the main loop if it is enabled in
 			// the config.
 			if config.Runtime.RPC.Enabled {
 				s := rpc.NewServer(":" + strconv.Itoa(config.Runtime.RPC.Port))
-				handler.Add("RPC", s.ListenAndServe, s.Close)
+				processHandler.Add("RPC", s.ListenAndServe, s.Close)
+			}
+
+			// Initialize ABI Service
+			processHandler.Add("ABI Service", services.AbiService.Run, services.AbiService.Close)
+
+			// Initialize State storage service
+			processHandler.Add("State Storage Service", services.StateService.Run, services.StateService.Close)
+
+			// Initialize Transactions Storage Service
+			processHandler.Add("Transactions Storage Service", services.TransactionService.Run, services.TransactionService.Close)
+
+			// Initialize Block Storage Service
+			processHandler.Add("Block Storage Service", services.BlockService.Run, services.BlockService.Close)
+
+			// Initialize Contract Hash storage service
+			database := db.Databaser(db.NewKeyValueDb(filepath.Join(config.Runtime.DbPath, "contractHash"), 0))
+			contractHashService := services.NewContractHashService(database)
+			processHandler.Add("Contract Hash Storage Service", contractHashService.Run, contractHashService.Close)
```

Comment:
Why is the `contractHashService` service initialized differently than the others? The other services are a single global variable for the whole project.

Reply:
Good catch. Thanks for mentioning that! I can refactor that in another PR once this branch is merged.